components/skus/browser/skus_url_loader_impl.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/skus/browser/skus_url_loader_impl.h"
+
+#include <utility>
+#include <vector>
+
+#include "brave/components/skus/browser/rs/cxx/src/lib.rs.h"
+#include "net/base/load_flags.h"
+#include "net/http/http_status_code.h"
+#include "services/network/public/mojom/url_response_head.mojom.h"
+#include "url/gurl.h"
+
+namespace skus {
+
+SkusUrlLoaderImpl::SkusUrlLoaderImpl(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : url_loader_factory_(url_loader_factory) {}
+
+SkusUrlLoaderImpl::~SkusUrlLoaderImpl() {}
+
+void SkusUrlLoaderImpl::BeginFetch(
+    const skus::HttpRequest& req,
+    rust::cxxbridge1::Fn<void(rust::cxxbridge1::Box<skus::HttpRoundtripContext>,
+                              skus::HttpResponse)> callback,
+    rust::cxxbridge1::Box<skus::HttpRoundtripContext> ctx) {
+  auto resource_request = std::make_unique<network::ResourceRequest>();
```

Comment:
since these appear to be intended for single use and can't handle multiple simultaneous requests, I would `DCHECK(!simple_url_loader_)` here

Reply:
Added in https://github.com/brave/brave-core/pull/10847/commits/f1fd81200a1006b5c0971b6805968acf4934c79a